benchmark/diagnostics_channel/publish.js

```diff
@@ -1,29 +1,50 @@
 'use strict';
 const common = require('../common.js');
-const dc = require('diagnostics_channel');
+const dc = require('node:diagnostics_channel');
 
 const bench = common.createBenchmark(main, {
   n: [1e8],
+  checkSubscribers: [1, 0],
+  objSize: [10, 1e2, 1e3],
   subscribers: [0, 1, 10],
 });
 
 function noop() {}
 
-function main({ n, subscribers }) {
+function createObj(size) {
+  return Array.from({ length: size }, (n) => ({
+    foo: 'yarp',
+    nope: {
+      bar: '123',
+      a: [1, 2, 3],
+      baz: n,
+      c: {},
+      b: [],
+    },
+  }));
+}
+
+function main({ n, subscribers, checkSubscribers, objSize }) {
   const channel = dc.channel('test');
   for (let i = 0; i < subscribers; i++) {
     channel.subscribe(noop);
   }
 
-  const data = {
-    foo: 'bar',
-  };
+  const data = createObj(objSize);
 
-  bench.start();
-  for (let i = 0; i < n; i++) {
+  const publishWithCheck = (data) => {
     if (channel.hasSubscribers) {
       channel.publish(data);
     }
+  };
+  const publishWithoutCheck = (data) => {
+    channel.publish(data);
+  };
+
+  const fn = checkSubscribers ? publishWithCheck : publishWithoutCheck;
+  bench.start();
+  for (let i = 0; i < n; i++) {
+    fn(data);
```

Comment:
This will have a call per iteration now. I would put the loop itself in each of the functions so there's nothing at all in the loop except exactly what is to be measured.

Reply:
I'm almost sure it gets inlined (the benchmark also shows the same result). But, I've changed it anyway. Did I get your suggestion right? 
834c602

Reply:
Yep, LGTM. I just prefer not to rely on VM magic to behave a particular way. Even if something is _probably_ the case right now, that could change at any time in the future. I find for benchmarks it's generally best to be very explicit about the desired behaviour to be sure the numbers don't change suddenly due to the influence of something _other_ than what the benchmark is intending to measure. 😅